src/sinks/util/builder.rs

```diff
@@ -100,13 +100,21 @@ pub trait SinkBuilderExt: Stream {
 
             Box::pin(async move {
                 // Split the input into metadata and events.
-                let (metadata, events) = builder.split_input(input);
+                let (metadata, request_metadata_builder, events) = builder.split_input(input);
```

Comment:
This is better. I'd suggest one more optimization. In every copy of `fn split_input`, there is the line:
```rust
    let builder = RequestMetadataBuilder::from(&events);
```
Could that be moved out here, probably _after_ calling `split_input` from what I see (ie on the split events rather than the source input)?

Reply:
Yeah, I did go down that path (spent a bit of time on it because have to redo that for all the sinks).

And basically I ran into the issue that I noted in my last commit https://github.com/vectordotdev/vector/pull/14836/commits/2921f9bca91a091f6f9a156936635aebc1392c28
, in builder.rs (because that's where I think this would get moved out to).

Was having a bear of a time fighting it and couldn't sort out the issue. Essentially moving that out makes each Event type for all the new style sinks have to implement all of the traits that any of the individual sink Event types require (eg `Serialize` , which created this weird conflicting trait implementation with the one in the JSON encoding stuff)


```
                // Note: it would be nice for the RequestMetadataBuilder to build be created from the
                // events here, and not need to be requred by split_input(). But this then requires
                // each Event type to implement Serialize, and that causes conflicts with the Serialize
                // implementation for EstimatedJsonEncodedSizeOf.
```